interface/main/finder/dynamic_finder_ajax.php

```diff
@@ -63,26 +109,26 @@
 // Global filtering.
 //
 $where = '';
+$sSearch = add_escape_custom(trim($_GET['sSearch']));
+$cnt_columns = count($aColumns);
+
 if (isset($_GET['sSearch']) && $_GET['sSearch'] !== "") {
-    $sSearch = add_escape_custom(trim($_GET['sSearch']));
-    foreach ($aColumns as $colname) {
-        $where .= $where ? "OR " : "WHERE ( ";
+    foreach ($aColumns as $index_col => $colname) {
+        $where .= $where ? ($cnt_columns != $index_col ? "OR " : "" ) : "WHERE ( ";
+
         if ($colname == 'name') {
-            if ($searchMethodInPatientList) { // exact search
-                $where .=
-                    "lname LIKE '$sSearch' OR " .
-                    "fname LIKE '$sSearch' OR " .
-                    "mname LIKE '$sSearch' ";
-            } else {
                 $where .= // like search
-                    "lname LIKE '$sSearch%' OR " .
-                    "fname LIKE '$sSearch%' OR " .
-                    "mname LIKE '$sSearch%' ";
-            }
-        } elseif ($searchMethodInPatientList) {
-            $where .= "`" . escape_sql_column_name($colname, array('patient_data')) . "` LIKE '$sSearch' ";
+                    "lname LIKE '".prepareMatchSearch($sSearch, "R", $searchMethodInPatientList)."' OR " .
+                    "fname LIKE '".prepareMatchSearch($sSearch, "R", $searchMethodInPatientList)."' OR " .
+                    "mname LIKE '".prepareMatchSearch($sSearch, "R", $searchMethodInPatientList)."' ";
         } else { // exact search
```

Comment:
is above `exact search` pertinent anymore?

Reply:
Can't this else be removed and changed to:
```
} else if ($colname == "DOB") {
etc.
```

Reply:
Everything remained as before, just slightly improved the code.